mkt/detail/views.py

```diff
@@ -26,16 +25,25 @@ def manifest(request, uuid):
     If not a packaged app, returns a 404.
 
     """
-    addon = get_object_or_404(Webapp, guid=uuid, is_packaged=True)
+    try:
+        addon = (Webapp.objects.only_translations()
+                 .get(disabled_by_user=False, is_packaged=True, guid=uuid))
+    except Webapp.DoesNotExist:
+        raise http.Http404
+
     is_avail = addon.status in [mkt.STATUS_PUBLIC, mkt.STATUS_UNLISTED,
                                 mkt.STATUS_BLOCKED]
-    is_owner = addon.authors.filter(pk=request.user.pk).exists()
-    is_owner_avail = addon.status == mkt.STATUS_APPROVED
+    if is_avail:
```

Comment:
Maybe rename those short variables now that we have more space ?

Like `is_available`, `is_available_to_owner` ?